Android/app/src/androidTest/java/io/github/project_travel_mate/EmptyEmailAndPasswordFieldsTest.java

```diff
@@ -0,0 +1,169 @@
+                    "android.permission.INTERNET",
+                    "android.permission.ACCESS_NETWORK_STATE",
+                    "android.permission.WAKE_LOCK",
+                    "android.permission.WRITE_CONTACTS",
+                    "android.permission.CAMERA",
+                    "android.permission.READ_EXTERNAL_STORAGE",
+                    "android.permission.BLUETOOTH",
+                    "android.permission.READ_PHONE_STATE");
+
+    @Test
+    public void emptyEmailAndPasswordFieldsTest() {
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        // Added a sleep statement to match the app's execution delay.
+        // The recommended way to handle such scenarios is to use Espresso idling resources:
+        // https://google.github.io/android-testing-support-library/docs/espresso/idling-resource/index.html
+        try {
+            Thread.sleep(700);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        ViewInteraction appCompatTextView = onView(
+                allOf(withId(R.id.signup), withText("Not registered yet? Create new account"),
+                        childAtPosition(
+                                childAtPosition(
+                                        withClassName(is("android.widget.LinearLayout")),
+                                        7),
+                                0),
+                        isDisplayed()));
+        appCompatTextView.perform(click());
+
+        ViewInteraction appCompatEditText = onView(
+                allOf(withId(R.id.input_firstname_signup),
+                        childAtPosition(
+                                childAtPosition(
+                                        withId(R.id.input_layout_firstname_signup),
+                                        0),
+                                0),
+                        isDisplayed()));
+        appCompatEditText.perform(replaceText("Arthur"), closeSoftKeyboard());
+
+        ViewInteraction appCompatEditText2 = onView(
+                allOf(withId(R.id.input_lastname_signup),
+                        childAtPosition(
+                                childAtPosition(
+                                        withId(R.id.input_layout_lastname_signup),
+                                        0),
+                                0),
+                        isDisplayed()));
+        appCompatEditText2.perform(replaceText("Wallace"), closeSoftKeyboard());
+
+        ViewInteraction flatButton = onView(
+                allOf(withId(R.id.ok_signup), withText("Signup"),
```

Comment:
can we have these string values as contants?
and comments one exactly what test are you validating?